### ISSUE TITLE
Is a Python 3.7 test running on travis-ci?

### ISSUE THREAD

**Issue by cdeil** [astropy/astropy#7772]
I wanted to add a Python 3.7 build on travis-ci for Gammapy a few days ago (see https://github.com/gammapy/gammapy/issues/1724) but was running into several issues with pytest (see  https://github.com/pytest-dev/pytest/issues/3854#issuecomment-415671196) and also Astropy.

Currently with Astropy master e.g.
```
pytest -v astropy/stats/lombscargle/tests/test_statistics.py
```
locally for me takes several minutes (?!) and then fails like this:
```
96 failed, 78 passed in 339.38 seconds
```

I wanted to understand this, and see if the issues also appear in the Astropy travis-ci builds.

As far as I can see, there's only this Python 3.7 build, and it's not running the tests!?
https://travis-ci.org/astropy/astropy/jobs/420715129#L480

Most of the other tests also don't seem to run the tests or a docs build and only take 1 minute:
https://travis-ci.org/astropy/astropy/builds/420715127

@bsipocz @astrofrog - Is there a problem with the Astropy travis-ci setup, or am I just not understanding how it works / looking at the wrong builds?

**Comment by bsipocz** [comment]
@cdeil - we may want to revisit which jobs to run from cron at some point, but the python 3.7 tests are indeed running on all PR commits and on master pushes, it's only disabled for the cron runs where we only run a subset of the jobs.

**Comment by bsipocz** [comment]
e.g. the last master run was here: https://travis-ci.org/astropy/astropy/builds/419745259

**Comment by drdavella** [comment]
I saw what @cdeil was talking about: it seems like a few consecutive test runs didn't actually run the tests at all. Maybe we can blame travis for a weird hiccup?

**Comment by bsipocz** [comment]
@drdavella - Both the links above are from cron runs, have you seen the tests not running in a normal run as well?

**Comment by drdavella** [comment]
Ah, no. Sorry for the noise. 🤐

**Comment by pllim** [comment]
So, I guess the answer is "yes except for cron jobs". Can we close this?

**Comment by cdeil** [comment]
Before we close: does a build exist on macOS with Python 3.7 that shows if it's working or not? If yes, can you share a link to a build log?

I get the error on macOS with Python 3.7 from conda, and there might be a real problem.

**Comment by cdeil** [comment]
I had a closer look. The problem I was seeing with
```
pytest -v astropy/stats/lombscargle/tests/test_statistics.py
```
was another incarnation as in #7769 of a pytest fixture being called and I got pytest 3.7 in that env.

The issue is here:

https://github.com/astropy/astropy/blob/41741c5a0c1133de4196157093834192eee66ad0/astropy/stats/lombscargle/tests/test_statistics.py#L113

That file has a fixture with parameters that is often used as a normal fixture. And then there's one case where Jake wanted different parameters so called it directly. @nicoddemus - how would you resolve that? Split the fixture into a `make_data` function with arguments, and then call that, and only use a fixture `data` which calls `make_data`?

See here:

https://github.com/astropy/astropy/blob/41741c5a0c1133de4196157093834192eee66ad0/astropy/stats/lombscargle/tests/test_statistics.py#L21-L31

It's not so useful, but here's the full low showing the issue and showing that it's gone when downdrading to pytest 3.6:
https://gist.github.com/cdeil/a504dd011282844f2460b30bdd42cd5f

Also: what are the chances that there are bugs discussed that involve both Python and pytest version numbers 3.6 and 3.7 exactly, that's just confusing.

**Comment by nicoddemus** [comment]
> Split the fixture into a make_data function with arguments, and then call that, and only use a fixture data which calls make_data?

Exactly. The problem of calling a fixture directly is that is often confusing for newcomers, that's why we added a deprecation warning which will be turned into an error in a future release. Splitting into a separate function is the recommended approach in this case. 👍

**Comment by cdeil** [comment]
Fixed in #7779 .

I'll close this issue now. The original description and discussion above is misleading and not useful. Sorry for that.

**Comment by bsipocz** [comment]
Thanks @cdeil! Though I have to disagree, the title may be misleading, but it's always useful to get reports for test failures.
